src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { rgb } from 'd3-color';
+import { scaleSequential } from 'd3-scale';
+import { interpolateMagma } from 'd3-scale-chromatic';
+
+export function computeTextureData(matrix: number[][]): Uint8Array {
+  // Flatten input matrix
+  const values = matrix.flat();
+
+  // Compute domain
+  const domain: [number, number] = values.reduce(
+    ([min, max], item) => [Math.min(item, min), Math.max(item, max)],
+    [Infinity, -Infinity]
+  );
+
+  // Generate D3 color map from domain
+  const colorMap = scaleSequential<string>(interpolateMagma);
+  colorMap.domain(domain);
+
+  // Compute RGB color for each datapoint `[[<r>, <g>, <b>], [<r>, <g>, <b>], ...]`
+  const colors = values.map(val => {
+    // Invoke color map and parse the CSS string it returns (`rgb(r, g, b)`)
+    const { r, g, b } = rgb(colorMap(val));
+    return [r, g, b];
+  });
+
+  // Prepare colors array for use as texture data
+  return Uint8Array.from(colors.flat());
+}
```

Comment:
Some comments are not really useful and can be removed.

Reply:
I reviewed the comments and extracted the min/max algorithm into its own function for self-documentation purposes.